src/AppInstallerRepositoryCore/Microsoft/SQLiteIndex.h

```diff
@@ -146,6 +146,12 @@ namespace AppInstaller::Repository::Microsoft
         // Largely a utility function; should not be used to do work on behalf of the index by the caller.
         Utility::NormalizedName NormalizeName(std::string_view name, std::string_view publisher) const;
 
+        // Get all the dependencies for a specific manifest.
+        std::map<Manifest::Dependency, SQLite::rowid_t> GetDependenciesByManifestRowId(SQLite::rowid_t manifestRowId) const;
+        std::vector<std::pair<Manifest::Manifest, Utility::Version>> GetDependenciesByPackageId(AppInstaller::Manifest::string_t packageId) const;

```

Comment:
```suggestion
        std::vector<std::pair<IdType, Utility::Version>> GetDependentsById(IdType id) const;
```
Where the pair is `{ Manifest ID, Version Required }`.  Really "highest version required" would be sufficient, but for reporting purposes we want to me able to call out the actual dependent.

Reply:
The actual dependent is the manifest returned by GetDependents as different manifests within the same package might not have dependencies on the current package,  which is why the manifest is returned for reporting sake. 

Also, there are two versions here, the manifest version is useful for reporting on the service and the min version associated with dependency relationship between the manifest and the package. Can you clarify what highest version required mean in this case?

Reply:
By that I mean that the only thing you really need to verify is that the highest version specified as a dependency will still be satisfied after a delete.  So you only need to know the highest version.

It is also interesting to know the set of manifests that would be impacted by the removal though, so returning the manifest id and the version of the dependency that it specifies allows you to determine all impacted manifests instead of just "is the set of impacted manifests empty?".

Reply:
The set of manifest that will be impacted is addressed in 215 - 232.

Reply:
Give this it's own comment that describes what the rowid is.
The input should be `IdType id` which is the package id's rowid.
I realize that it makes it harder given the way the DependencyGraph works now, but that should be fixed, not this made to conform.